### ISSUE TITLE
关于torch.mean(torch.sum(loss))的问题

### ISSUE THREAD

**Issue by ezio0218** [Forrest-Stone/MA-GNN#2]
您好，我主要想问一下为什么在这个地方
![image](https://user-images.githubusercontent.com/46940734/173008231-e0def87f-0e78-4116-9ea1-5275c48ad214.png)
要加torch.mean，打印出来好像都是一样的呀。sum的时候好像就已经没有维度了

**Comment by Forrest-Stone** [comment]
hello, sum(loss) 里面的loss跟预测的target个数有关

**Comment by ezio0218** [comment]
![image](https://user-images.githubusercontent.com/46940734/173010783-9645e5de-af9a-4137-923d-f8594d157375.png)
![image](https://user-images.githubusercontent.com/46940734/173010858-5b17ba9d-fb9f-4daa-87e0-c25d17ee691e.png)
感谢您的回复，就以实例中的target和neg都是2，batch_size为4096为例。pos_score和neg_score的维度此时都是[4096,2]，所以您做torch.mean(torch.sum())的想法是希望两个target的loss相加然后除以batch_size的大小吗？如果直接求mean()是可以的嘛

**Comment by ezio0218** [comment]
主要是我看伯乐库中对于BPR的计算是直接求的mean：
![image](https://user-images.githubusercontent.com/46940734/173012909-473d1410-e71a-4b71-9480-60065cd65210.png)
当然一般情况下带有label的数据target大小都是1，在您的数据场景下target会大于1

**Comment by Forrest-Stone** [comment]
> ![image](https://user-images.githubusercontent.com/46940734/173010783-9645e5de-af9a-4137-923d-f8594d157375.png) ![image](https://user-images.githubusercontent.com/46940734/173010858-5b17ba9d-fb9f-4daa-87e0-c25d17ee691e.png) 感谢您的回复，就以实例中的target和neg都是2，batch_size为4096为例。pos_score和neg_score的维度此时都是[4096,2]，所以您做torch.mean(torch.sum())的想法是希望两个target的loss相加然后除以batch_size的大小吗？如果直接求mean()是可以的嘛

hello，主要是涉及到batch_size是否加入计算，我觉得都是可以的

**Comment by ezio0218** [comment]
好的 非常感谢及时的回复！最后想问一下，如果想用pyg库中写好的gnn layer来替换magnn中的GNNCell模块功能是可行的吗

**Comment by Forrest-Stone** [comment]
是可行的，功能对应着就可以
